python/lsst/ts/pmd/component.py

```diff
@@ -165,6 +164,7 @@ async def get_slots_position(self):
 
         if not self.connected:
             raise Exception("Not connected")
+        isok = None
```

Comment:
You don't need this variable, as you set it just before you return it in both cases. It would make the code more readable to drop it and return an explicit False or True.

Reply:
Also I *strongly* suggest initializing `position` using:
```
position = [math.nan] * len(self.names)
```
or similar (e.g. if you have a constant or variable with the number of channels, you can use that instead of len(self.names)). Advantages:
* It is obvious to the reader *why* the list is the length it is
* It scales well if you change the # of channels
* You don't have to count lines of math.nan to make the code right

Reply:
Fix the doc string for this function to describe the new isok return value.

Also, please consider documenting that it does not reset the multiplexor